charts/argo-cd/templates/argocd-repo-server/servicemonitor.yaml

```diff
@@ -29,8 +24,6 @@ spec:
       - {{ .Release.Namespace }}
   selector:
     matchLabels:
-      app.kubernetes.io/instance: {{ .Release.Name }}
-      app.kubernetes.io/name: {{ template "argo-cd.repoServer.fullname" . }}-metrics
-      app.kubernetes.io/component: {{ .Values.repoServer.name }}
+      {{- include "argo-cd.selectorLabels" (dict "context" . "component" .Values.repoServer.name "name" (printf "%s-metrics" .Values.repoServer.name)) | nindent 6 }}
```

Comment:
See comment: https://github.com/argoproj/argo-helm/pull/627/files#r596800744
luckily this selector is mutable :-)